src/twisted/application/runner/_pidfile.py

```diff
@@ -8,59 +8,60 @@
 
 import errno
 from os import getpid, kill, name as SYSTEM_NAME
+from types import TracebackType
+from typing import Optional, Type
 
 from zope.interface import Interface, implementer
 
 from twisted.logger import Logger
+from twisted.python.filepath import FilePath
 
 
 class IPIDFile(Interface):
```

Comment:
Since this isn't public API yet… question: should this be a [Protocol](https://mypy.readthedocs.io/en/stable/protocols.html?highlight=__exit__) or stay as a Zope Interface?

@glyph IIRC you said Zope last time I asked, but want to make sure before we publish this API.

Reply:
I still have a slight preference for Interface given the fact that many of these methods have quite generic names.  (I mean, it'd be hard to implement `writeRunningPID` by accident, but not impossible.)  But if anyone else has a strong opinion here I would be OK with either.  I don't see any `.providedBy` checks so either kind of works.

Thanks for checking in :-).

Reply:
The main argument for `Protocol` is using the stdlib, but it's not like that's gong to change much for Twisted.  I'll leave it alone.